packages/data-context/src/DataContext.ts

```diff
@@ -153,8 +154,13 @@ export class DataContext {
     return new GitDataSource(this)
   }
 
-  async versions () {
-    return new VersionsDataSource().versions()
+  get versions () {
+    // Cache versionsDataSource this way so that we can stub it in tests
+    if (!this._versionsDataSource) {
+      this._versionsDataSource = new VersionsDataSource(this)
+    }
```

Comment:
Hmm, what was the issue w/ testing that we can't use `@cached`?

Reply:
We can't swizzle versions like this: https://github.com/cypress-io/cypress/pull/20107/files#diff-6d8838def800c5d83c3b41191d48f8d33dbffc76c923af2e76d3b0bd94a51774R108-R124

Reply:
Nevermind. Have a fix for that.

Reply:
[`8220bed` (#20107)](https://github.com/cypress-io/cypress/pull/20107/commits/8220bed45b43a8e12d7e00e1fdfc4f7f38b93d54)